gpu/benches/benchmarks/metal.rs

```diff
@@ -1,41 +1,49 @@
 use criterion::Criterion;
-use lambdaworks_gpu::metal::{abstractions::state::MetalState, fft::ops::*};
+use lambdaworks_gpu::metal::{
+    abstractions::state::MetalState,
+    fft::{ops::*, polynomial::*},
+};
 use lambdaworks_math::{
     field::{element::FieldElement, traits::IsTwoAdicField},
-    field::{test_fields::u32_test_field::U32TestField, traits::RootsConfig},
+    field::{fields::fft_friendly::stark_252_prime_field::Stark252PrimeField, traits::RootsConfig},
+    polynomial::Polynomial,
+    unsigned_integer::element::UnsignedInteger,
 };
 use rand::random;
 
-type F = U32TestField;
+type F = Stark252PrimeField;
 type FE = FieldElement<F>;
+const INPUT_SET: [u64; 4] = [21, 22, 23, 24];
 
-fn gen_coeffs(order: usize) -> Vec<FE> {
+fn gen_coeffs(order: u64) -> Vec<FE> {
     let mut result = Vec::with_capacity(1 << order);
     for _ in 0..result.capacity() {
-        result.push(FE::new(random()));
+        let rand_big = UnsignedInteger { limbs: random() };
+
+        result.push(FE::new(rand_big));
     }
     result
 }
 
 pub fn metal_fft_benchmarks(c: &mut Criterion) {
-    let mut group = c.benchmark_group("FFT");
+    let mut group = c.benchmark_group("Ordered FFT");
+    group.sample_size(10); // too slow otherwise
 
-    for order in 21..=24 {
+    for order in INPUT_SET {
         let coeffs = gen_coeffs(order);
-        group.throughput(criterion::Throughput::Elements(1 << order)); // info for criterion
+        group.throughput(criterion::Throughput::Elements(1 << order));
 
         // the objective is to bench ordered FFT, including twiddles generation and Metal setup
         group.bench_with_input(
-            format!("Parallel NR radix2 FFT for 2^{order} elements"),
+            "Parallel from NR radix2 (Metal)",
             &coeffs,
             |bench, coeffs| {
                 bench.iter(|| {
                     // TODO: autoreleaspool hurts perf. by 2-3%. Search for an alternative
                     objc::rc::autoreleasepool(|| {
                         let coeffs = coeffs.clone();
                         let metal_state = MetalState::new(None).unwrap();
-                        let twiddles =
-                            F::get_twiddles(order as u64, RootsConfig::BitReverse).unwrap();
+                        let twiddles = F::get_twiddles(order, RootsConfig::BitReverse).unwrap();
 
                         fft(&coeffs, &twiddles, &metal_state).unwrap();
                     });
```

Comment:
We can use the function to evaluate directly from the polynomial to be the same as the benchmark for interpolation. Create a polynomial from the coeffs before the benchmark starts and measure `poly.evaluate_fft_metal(&metal_state)` that already has the twiddle generation.

Reply:
I would propose to add a different benchmark for polynomial evaluation, and strip this one from twiddle gen and leave only ordered FFT, so each implementation (Metal, sequential and in the near future CUDA and WebGPU) can be compared easily. wdyt?

Reply:
Yeah, seems like a good idea

Reply:
Done!